src/civetweb.c

```diff
@@ -16662,7 +16692,16 @@ free_context(struct mg_context *ctx)
 #ifndef NO_SSL
 	/* Deallocate SSL context */
 	if (ctx->ssl_ctx != NULL) {
+	  SSL_CTX* ssl_ctx = 0;
+      int callback_ret =
+	    (ctx->callbacks.get_external_ssl_ctx == NULL)
+	        ? 0
+	        : (ctx->callbacks.get_external_ssl_ctx(&ssl_ctx, ctx->user_data));
```

Comment:
Review comment - init/exit:

`SSL_CTX* ssl_ctx = 0;`
`get_external_ssl_ctx(&ssl_ctx, ..)` 
called for both, init and exit.
Suggestion: In the exit case, use
`SSL_CTX* ssl_ctx = ctx->ssl_ctx;`
`get_external_ssl_ctx(&ssl_ctx, ..)`
So the callback can distinguish between init and exit by checking if `*ssl_ctx` is NULL (init) or not (exit).

Minor: Then "get_" is not the ideal name for the operation, since "exit" situation is not exactly a "get" but a "done with it".